appcues/src/main/java/com/appcues/Storage.kt

```diff
@@ -21,31 +25,85 @@ internal class Storage(
         context.getSharedPreferences("com.appcues.storage.${config.applicationId}", Context.MODE_PRIVATE)
 
     var deviceId: String
```

Comment:
I repeated the same logic for all propertyes, basically created a private _xxxxx: T property we get get() from.

on set we update this value + update it on sharedPreference (following previous rules you had in place for each)

one thing I tried to do was to refactory this "block" of code into a property delegate, but it ended up being a lot more code, more complex than it should be, so this is the just the simple fix.

https://kotlinlang.org/docs/delegated-properties.html